### ISSUE TITLE
AttributeError: 'GPTVectorStoreIndex' object has no attribute 'query'

### ISSUE THREAD

**Issue by geg00** [jerryjliu/llama_index#2497]
(.venv) C:\Users\hbb9279\OpenAI\Files\KBR>python
Python 3.11.3 (tags/v3.11.3:f3909b8, Apr  4 2023, 23:49:59) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import llama_index
>>> llama_index.__version__
'0.6.0'

 cur_index = GPTVectorStoreIndex.from_documents(doc_set[file], service_context=service_context, storage_context=storage_context)
 cur_index.index_struct.index_id="Index1"
 print(cur_index.query("Summarize this"))

  File "C:\Users\hbb9279\OpenAI\code\bot.py", line 127, in <module>
    print(cur_index.query("Summarize this"))
          ^^^^^^^^^^^^^^^
AttributeError: 'GPTVectorStoreIndex' object has no attribute 'query'

**Comment by sumanboda** [comment]
>>> index = GPTVectorStoreIndex.from_documents(documents)
>>> prompt="what is skhokho?"
>>> r=index.query(prompt)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'GPTVectorStoreIndex' object has no attribute 'query'

**Comment by adnanafzalbhatti** [comment]
Use this:
# Querying the index
query_engine = index.as_query_engine()
response = query_engine.query("what are the articles about?")
print(response)

Source: https://gpt-index.readthedocs.io/en/latest/guides/primer/usage_pattern.html#optional-save-the-index-for-future-use

**Comment by shobhitag11** [comment]
It didn't work for me.

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-35-874f59c8bded>](https://localhost:8080/#) in <cell line: 4>()
      2 
      3 query_engine = index.as_query_engine()
----> 4 response = query_engine.query("What are the top 10 players?")

18 frames
[/usr/local/lib/python3.10/dist-packages/langchain/llms/base.py](https://localhost:8080/#) in dict(self, **kwargs)
    335     def dict(self, **kwargs: Any) -> Dict:
    336         """Return a dictionary of the LLM."""
--> 337         starter_dict = dict(self._identifying_params)
    338         starter_dict["_type"] = self._llm_type
    339         return starter_dict

TypeError: 'method' object is not iterable

**Comment by AmlanSamanta** [comment]
Same issue I also faced in below snippet:

query_engine = index.as_query_engine()
response = query_engine.query("Who is the author of this article?")

TypeError: 'method' object is not iterable

**Comment by tractorjuice** [comment]
I managed to fix the error in my code.
I had to add:
        query_engine = index.as_query_engine()

and change the query to:
        response = query_engine.query(prompt)

Working source code here: https://github.com/tractorjuice/wardley-qa-video/blob/main/wardley-qa-video.py

**Comment by VirajOke** [comment]
I have the same error. It was working fine before the changes were made to the querying structure.
```
query_engine = index.as_query_engine()
response = query_engine.query("can you summarize the documents?")
```
`TypeError: 'method' object is not iterable `

**Comment by Marinos96** [comment]
> It didn't work for me.
> 
> TypeError Traceback (most recent call last) in <cell line: 4>() 2 3 query_engine = index.as_query_engine() ----> 4 response = query_engine.query("What are the top 10 players?")
> 
> 18 frames [/usr/local/lib/python3.10/dist-packages/langchain/llms/base.py](https://localhost:8080/#) in dict(self, **kwargs) 335 def dict(self, **kwargs: Any) -> Dict: 336 """Return a dictionary of the LLM.""" --> 337 starter_dict = dict(self._identifying_params) 338 starter_dict["_type"] = self._llm_type 339 return starter_dict
> 
> TypeError: 'method' object is not iterable

@shobhitag11 I guess you are using a custom llm via langchain. If yes, make sure to comply with the latest changes in the LLM class. --> https://python.langchain.com/en/latest/modules/models/llms/examples/custom_llm.html
`_llm_type` & `_identifying_params` must be defined as properties (`@property`)  in the wrapper class.

After applying these changes, `query` works again for me.

**Comment by ANANAYAGG** [comment]
actually i am using google flan t5 in custom llm  ---
code is--


class customLLM(LLM):
    model_name = "google/flan-t5-large"
    pipeline = pipeline("text2text-generation", model=model_name, device=0, model_kwargs={"torch_dtype":torch.bfloat16})

    def _call(self, prompt, stop=None):
        return self.pipeline(prompt, max_length=9999)[0]["generated_text"]
 
    def _identifying_params(self):
        return {"name_of_model": self.model_name}

    def _llm_type(self):
        return "custom"


llm_predictor = LLMPredictor(llm=customLLM())


can you please guide how to solve

 query_engine = index.as_query_engine()
response = query_engine.query("can you summarize the documents?")
TypeError: 'method' object is not iterable
this error

**Comment by sarikasarika0750** [comment]
I m using flan t5 too, i m facing the same error, someone please help me with this.

**Comment by sarikasarika0750** [comment]
query_engine = index.as_query_engine()
response = query_engine.query("how many parameters does the BloombergGPT contain?")
return response

I m using flan t5 too, but it doesn't seem to work out.

**Comment by anguswilliams93** [comment]
I get:

ImportError: cannot import name 'EventPayload' from 'llama_index.callbacks.schema'
